pennylane/devices/qubit/sampling.py

```diff
@@ -307,7 +282,18 @@ def _process_single_shot(samples):
             raise e
         samples = qml.math.full((shots.total_shots, len(wires)), 0)
 
-    return _process_single_shot(samples)
+    processed_samples = []
+    for lower, upper in shots.bins():
+        if len(samples.shape) == 3:
```

Comment:
```suggestion
        if is_state_batched:
```

Both work, but might as well re-use the pre-calculated variable.